mne/preprocessing/_css.py

```diff
@@ -0,0 +1,115 @@
+# Author: John Samuelsson <johnsam@mit.edu>
+
+import numpy as np
+from ..utils import logger
+from ..evoked import EvokedArray, Evoked
+from .. import pick_types
+from numpy import linalg
+
+
+def temp_proj(ref_2, ref_1, raw_data, n_proj=6):
+    """Removes common signal subspace of ref_2 and ref_1 from raw_data.
+
+    Parameters
+    ----------
+    ref_2 : np.ndarray of float, shape (n_sensors_2, n_times)
+        The magnetometer data for CSS. Can use either all magnetometer data or
+        a few selected sensors close to a region to be suppressed.
+    ref_1 : np.ndarray of float, shape (n_sensors_1, n_times)
+        The gradiometer data for CSS. Can use either all gradiometer data or
+        a few selected sensors close to a region to be suppressed.
+    raw_data : np.ndarray of float, shape (n_sensors_raw, n_times)
+        The data to be filtered, typically the EEG data.
+    n_proj : int
+        The number of projection vectors.
+
+    Returns
+    -------
+    filtered_data : np.ndarray of float, shape (n_sensors_raw, n_times)
+        The filtered data.
+
+    Notes
+    -------
+    This temporal projection procedure removes the common signal subspace
+    between ref_2 and ref_1 from raw_data using n_proj number of
+    projection vectors. Normally used for cortical signal suppression, where
+    ref_1 is gradiometer data, ref_2 is magnetometer data and
+    raw_data is EEG data.
+
+    """
+    # Orthonormalize gradiometer and magnetometer data by a QR decomposition
+    Qg = linalg.qr(ref_1.T)[0]
+    Qm = linalg.qr(ref_2.T)[0]
+
+    # Calculate cross-correlation
+    C = np.dot(Qg.T, Qm)
+
+    # Channel weights for common temporal subspace by SVD of cross-correlation
+    Y, S, Z = linalg.svd(C)
+
+    # Get temporal signals from channel weights
+    u = np.dot(Qg, Y)
+
+    # Project out common subspace
+    filtered_data = raw_data
+    for i in range(n_proj):
+        proj_vec = u[:, i].reshape(u.shape[0], 1)
+        weights = np.dot(filtered_data, proj_vec)
+        filtered_data = filtered_data - np.dot(weights, proj_vec.T)
+
+    return filtered_data
+
+
+def cortical_signal_suppression(evoked, n_proj=6):
```

Comment:
Since the underlying function mentions the use case of passing just a few sensors near the site of the activity to be suppressed, WDYT about parameters `picks_mag` and `picks_grad` here?  You'd want a corresponding warning in the notes section that the sensors should be co-located, and possibly even a check in the code to make sure that's the case...

Reply:
Good idea, I created mag_inds and grad_inds as optional arguments; if they are not specified, the default all channels will be used. The picked channels do not necessarily have to be at the same locations so I left that open. Likely, vast majority of users will use the default option of using all channels anyway